patchwork-gui/src/main/java/net/patchworkmc/mixin/gui/MixinMinecraftClient.java

```diff
@@ -0,0 +1,116 @@
+
+package net.patchworkmc.mixin.gui;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.At.Shift;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.ModifyVariable;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import net.minecraftforge.client.event.GuiOpenEvent;
+import net.minecraftforge.common.MinecraftForge;
+
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.gui.screen.OutOfMemoryScreen;
+import net.minecraft.client.gui.screen.Screen;
+
+@Mixin(MinecraftClient.class)
+public class MixinMinecraftClient {
+	private static final String PATCHWORK_YARN_MTD_OPENSCREEN = "openScreen(Lnet/minecraft/client/gui/screen/Screen;)V";
+	private static final String PATCHWORK_YARN_MTD_SCREEN_REMOVE = "net/minecraft/client/gui/screen/Screen.removed()V";
+
+	// net.minecraft.client.gui.TitleScreen --> net.minecraft.class_442
+	private static final String PATCHWORK_YARN_CLS_TITLESCREEN = "classValue=net/minecraft/client/gui/screen/TitleScreen";
+	private static final String PATCHWORK_REOBF_CLS_TITLESCREEN = "classValue=net/minecraft/class_442";
+
+	private static final Screen PATCHWORK_GUIOPENEVENT_CANCEL_MAKRER = new OutOfMemoryScreen();
+	private static Screen patchwork_oldScreen;
+
+	@Shadow
+	public Screen currentScreen;
+
+	@Redirect(method = PATCHWORK_YARN_MTD_OPENSCREEN, at = @At(value = "INVOKE", target = PATCHWORK_YARN_MTD_SCREEN_REMOVE))
+	private void patchwork_suppressScreenRemoved(Screen screen) {
+		// No-op (handled in patchwork_fireOpenEvent)
+	}
+
+	@ModifyVariable(method = PATCHWORK_YARN_MTD_OPENSCREEN, at = @At(value = "CONSTANT", args = PATCHWORK_YARN_CLS_TITLESCREEN, shift = Shift.BY, by = -2), require = 0)
+	public Screen patchwork_yarn_fireOpenEvent(Screen screen) {
+		return patchwork_impl_fireOpenEvent(screen);
+	}
+
+	@ModifyVariable(method = PATCHWORK_YARN_MTD_OPENSCREEN, at = @At(value = "CONSTANT", args = PATCHWORK_REOBF_CLS_TITLESCREEN, shift = Shift.BY, by = -2), require = 0)
+	public Screen patchwork_reobf_fireOpenEvent(Screen screen) {
+		return patchwork_impl_fireOpenEvent(screen);
+	}
+
+	@Inject(method = PATCHWORK_YARN_MTD_OPENSCREEN, at = @At(value = "CONSTANT", args = PATCHWORK_YARN_CLS_TITLESCREEN), cancellable = true, require = 0)
+	public void patchwork_yarn_cancelOpening(Screen screen, CallbackInfo callback) {
+		patchwork_impl_cancelOpening(screen, callback);
+	}
+
+	@Inject(method = PATCHWORK_YARN_MTD_OPENSCREEN, at = @At(value = "CONSTANT", args = PATCHWORK_REOBF_CLS_TITLESCREEN), cancellable = true, require = 0)
+	public void patchwork_reobf_cancelOpening(Screen screen, CallbackInfo callback) {
+		patchwork_impl_cancelOpening(screen, callback);
+	}
+
+	/**
+	 * Sets the argument Screen as the main (topmost visible) screen.
+	 * <br>
```

Comment:
Nitpick: fabric style is `<p>`, use that here.

Reply:
fixed